modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustClientCodegen.java

```diff
@@ -455,38 +448,16 @@ public String getSchemaType(Schema p) {
         String schemaType = super.getSchemaType(p);
         String type = typeMapping.getOrDefault(schemaType, schemaType);
 
-        boolean bestFit = convertPropertyToBoolean(BEST_FIT_INT);
-        boolean unsigned = convertPropertyToBoolean(PREFER_UNSIGNED_INT);
-
-        if (bestFit || unsigned) {
-            BigDecimal maximum = p.getMaximum();
-            BigDecimal minimum = p.getMinimum();
-
-            try {
-              if (maximum != null && minimum != null) {
-                  long max = maximum.longValueExact();
-                  long min = minimum.longValueExact();
-
-                  if (unsigned && bestFit && max <= (Byte.MAX_VALUE * 2) + 1 && min >= 0) {
-                      type = "u8";
-                  } else if (bestFit && max <= Byte.MAX_VALUE && min >= Byte.MIN_VALUE) {
-                      type = "i8";
-                  } else if (unsigned && bestFit && max <= (Short.MAX_VALUE * 2) + 1 && min >= 0) {
-                      type = "u16";
-                  } else if (bestFit && max <= Short.MAX_VALUE && min >= Short.MIN_VALUE) {
-                      type = "i16";
-                  } else if (unsigned && bestFit && max <= (Integer.MAX_VALUE * 2L) + 1 && min >= 0) {
-                      type = "u32";
-                  } else if (bestFit && max <= Integer.MAX_VALUE && min >= Integer.MIN_VALUE) {
-                      type = "i32";
-                  }
-              }
-            } catch (ArithmeticException a) {
-                // no-op; case will be handled in the next if block
-            }
-
-            if (unsigned && minimum != null && minimum.compareTo(BigDecimal.ZERO) >= 0 && unsignedMapping.containsKey(type)) {
-                type = unsignedMapping.get(type);
+        if (Objects.equals(p.getType(), "integer")) {
+            boolean bestFit = convertPropertyToBoolean(BEST_FIT_INT);
+            boolean preferUnsigned = convertPropertyToBoolean(PREFER_UNSIGNED_INT);
+            if (bestFit || preferUnsigned) {
+                return bestFittingIntegerType(
```

Comment:
this isn't quite right - we should only bestFit if bestFit is set - will fix in a bit